### ISSUE TITLE
Extracting softmax outputs for all LSTM units so as to get continuous prediction of sequential data

### ISSUE THREAD

**Issue by krayush07** [tflearn/tflearn#205]
In the given segment of code:

```
input_net = tflearn.input_data([None, 4, 3])
lstm_net = tflearn.layers.recurrent.lstm(input_net, 10, dropout=0.8, name='lstm_layer', return_seq=True)
softmax_net = tflearn.fully_connected(lstm_net[3], 3, activation='softmax', name='softmax_layer') #lstm_net[3] because error propagation is on the final prediction
regression_net = tflearn.regression(softmax_net, optimizer='adam', learning_rate=0.001, loss='categorical_crossentropy')
```
How can I extract the softmax predictions for _all LSTM units_? 

I am currently using model.predict() method to extract the output, however I want to extract the softmax outputs for all the LSTM units instead.

**Comment by aymericdamien** [comment]
Do you mean for every time step? You could do something like that:
```python
input_net = tflearn.input_data([None, 4, 3])
lstm_net = tflearn.layers.recurrent.lstm(input_net, 10, dropout=0.8, name='lstm_layer', return_seq=True)
# We set a softmax per timestep and as many optimizers
outputs = [tflearn.regression(tflearn.fully_connected(ts, 3, activation='softmax', name='softmax_layer', optimizer='adam', learning_rate=0.001, loss='categorical_crossentropy')) for ts in lstm_net]
# Here you can regroup your results (prediction at each timestep)
net = tflearn.merge(outputs, axis=1)
```

**Comment by krayush07** [comment]
Great. Thanks.

`net = tflearn.merge(outputs, axis=1)` isn't working. It says `_TypeError: merge() takes at least 2 arguments (2 given)_`

Do you mean tflearn.merge_outputs (outputs, name='MergeOutputs')?

Also Y is vectorial representation of labels i.e, [[0,1,0,0,0], [0,1,0,0,0], [0,0,0,1,0], [1,0,0,0,0]] for classes 2, 2, 4 and 1 respectively, right?

**Comment by aymericdamien** [comment]
yes, you can use 'merge_outputs', and Y are indeed your labels.

**Comment by krayush07** [comment]
Okay. I am, however, running into `_index-out of range error_`. X is my input matrix of shape [None, 4, 3], 4 is input size and 3 is feature length. Y is a single integer. So [Y, Y, Y, Y] = [2, 2, 4, 1].  The issue arises in `utils.py: net_targets[i]`. What is the correct way to feed the training data and the labels for each LSTM unit in model.fit() ?

**Comment by aymericdamien** [comment]
Y should be [[0,1,0,0,0], [0,1,0,0,0], [0,0,0,1,0], [1,0,0,0,0]], according to your example.
Then, you have to feed a label per regression layer you created.
So you could replace your last line by:
```python
model.fit(X, [Y for i in len(outputs)])
```

**Comment by krayush07** [comment]
@aymericdamien Yes, I am doing it that way.

`model.fit(trainInput, [[trainLabel[i][j] for j in range(len(outputs))] for i in range(len(trainLabel))])`

trainInput = Input Matrix of 8x4x3
trainLabel = Binary matrix representing labels for each LSTM unit. Output dimesion (possible labels) is 3
So a trainLabel[i][j] would look like:
[array([ 0.,  0.,  1.]), array([ 0.,  1.,  0.]), array([ 1.,  0.,  0.]), array([ 0.,  1.,  0.])]

However, I am still getting error while setting the feed_dict[net_targets[i]] = y in utils.py.

What am I still missing?
